### PR TITLE
Client dir compatibility, part 1

### PR DESCRIPTION
**Pull request by tigercat2000**
This commit modifies all of our smoothing sprites to work properly when
the client's screen is rotated. It also does the same for the lighting
overlays, adding directional sprites so that they rotate correctly.

It also ports the movement input system from Baystation, which means
that your controls aren't suddenly reversed when your screen is rotated.

### New

You can see that a lot of stuff is still broken- this is still a fair way away from allowing everyone to freely rotate their client without major bugs.

### CODE REVIEW

**Comment by taukausanake** [comment]
This is nice, but you're not going to _actively_ rotate your client much, are you? This is just for the shuttle rotation, right? _Right?_

I'd rather not have rotation than broken anything. Also it's REALLY fucking with my brain seeing those screenshots and that even if I don't notice the broken sprites

**Comment by granodd** [comment]
does still need to do something about that bar sign. Looks nice, and is also kinda trippy to think of the station from these different perspectives

**Comment by tigercat2000** [comment]
@taukausanake It's not for anything right now, it's a feature of 511- it'd be unwise to forcefully change the client's direction at this time. I'm not sure what you're on about with shuttle rotation, it has nothing to do with that.

Ideally, we'd eventually have the same as Polaris/Bay does- preference verbs to freely rotate your client. It's an interesting thing to play with, if nothing else.

@granodd Yeah, the bar sign (and all 32x64) sprites are buggy as hell. It's a bit of a pain to fix them.

**Comment by Regen1** [comment]
@tigercat2000 I'd much rather have rotation of the clientdir be an effect of a drug of some kind, it's amazing at disorienting the player, and goes great lengths at making the game more immersive.

Lifeweb being an example, cave "distortion" would simply rotate your client view, and it made navigating the caves extremely hard, having that as a drug effect would be really cool.

**Comment by tigercat2000** [comment]
@Regen1 Intentionally disorienting the player is questionable, and either way, it's something for a later PR

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -17,26 +17,29 @@
...
-	Move(get_step(mob, EAST), EAST)
+	DO_MOVE(EAST)
 
 /client/verb/moveleft()
 	set name = ".moveleft"
 	set instant = 1
-	Move(get_step(mob, WEST), WEST)
+	DO_MOVE(WEST)
 
+#undef DO_MOVE
```
Comment (Crazylemon64) [inline code/modules/mob/mob_movement.dm]: As a comment to lay out my understanding of the changes of this PR:
The changes to this file cause client movement inputs to be appropriately rotated depending on the rotation of the client's view.

**Pull request merged**